scipy/optimize/_linprog_util.py

```diff
@@ -1111,9 +1063,9 @@ def vstack(blocks):
     l_nolb_someub = np.logical_and(lb_none, ub_some)
     i_nolb = np.nonzero(l_nolb_someub)[0]
     lbs[l_nolb_someub], ubs[l_nolb_someub] = (
-        -ubs[l_nolb_someub], lbs[l_nolb_someub])
-    lb_none = np.equal(lbs, None)
-    ub_none = np.equal(ubs, None)
+        -ubs[l_nolb_someub], -lbs[l_nolb_someub])
```

Comment:
Same here - can you remind me what was happening and why it is changing sign?

Reply:
This is where a variable with an upper bound only is converted into a variable with a lower bound only, by substituting `-x` for `x`. The bounds change from `(-inf, V)` into `(-V, inf)`, so reverse lower and upper bound, and invert both.
The original code only needed to invert the upper bound: `(None, V)` into `(-V, None)`, so did not include 2nd minus sign.
It took some time before I understood this :-)

Reply:
OK! I'll take your word for it. I wrote it and I forgot, so maybe comments would help the next reader : )